src/Symfony/Component/DependencyInjection/Compiler/AutowireRequiredPropertiesPass.php

```diff
@@ -49,6 +51,8 @@ protected function processValue(mixed $value, bool $isRoot = false): mixed
                 continue;
             }
 
+            trigger_deprecation('symfony/dependency-injection', '6.3', 'The "@required" annotation will be removed. Use the attribute "Symfony\Contracts\Service\Attribute\Required" instead.');
```

Comment:
Let's move this before the array_key_exists +

will be removed. Use => is deprecated, use